tool/tctl/common/access_request_command.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2015-2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gravitational/kingpin"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/asciitable"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/trace"
+)
+
+// AccessRequestCommand implements `tctl users` set of commands
+// It implements CLICommand interface
+type AccessRequestCommand struct {
+	config *service.Config
+	reqIDs []string
+
+	user  string
+	roles string
+	// format is the output format, e.g. text or json
+	format string
+
+	requestList    *kingpin.CmdClause
+	requestApprove *kingpin.CmdClause
+	requestDeny    *kingpin.CmdClause
+	requestCreate  *kingpin.CmdClause
+	requestDelete  *kingpin.CmdClause
+}
+
+// Initialize allows AccessRequestCommand to plug itself into the CLI parser
+func (c *AccessRequestCommand) Initialize(app *kingpin.Application, config *service.Config) {
+	c.config = config
+	requests := app.Command("requests", "Manage access requests").Alias("request")
+
+	c.requestList = requests.Command("ls", "Show active access requests")
+	c.requestList.Flag("format", "Output format, 'text' or 'json'").Hidden().Default(teleport.Text).StringVar(&c.format)
+
+	c.requestApprove = requests.Command("approve", "Approve pending access request")
```

Comment:
So this will be 

`tctl request approve [request-id]` ( just a note for myself as it wasn't in the description.